stake-pool/js/src/schema.ts

```diff
@@ -0,0 +1,147 @@
+import {Schema, serialize, deserializeUnchecked} from 'borsh';
+import BN from 'bn.js';
+import {SOLANA_SCHEMA, Struct, Enum} from '@solana/web3.js';
+
+export class Fee extends Struct {
+  denominator: BN;
+  numerator: BN;
+}
+
+export class AccountType extends Enum {}
+
+export class AccountTypeEnum extends Struct {}
+
+export enum AccountTypeKind {
+  Uninitialized = 'Uninitialized',
+  StakePool = 'StakePool',
+  ValidatorList = 'ValidatorList',
+}
+
+export class StakePool extends Struct {
+  accountType: AccountType;
+  manager: PublicKey;
+  staker: PublicKey;
+  depositAuthority: PublicKey;
+  withdrawBumpSeed: number;
+  validatorList: PublicKey;
+  reserveStake: PublicKey;
+  poolMint: PublicKey;
+  managerFeeAccount: PublicKey;
+  totalStakeLamports: BN;
+  poolTokenSupply: BN;
+  lastUpdateEpoch: BN;
+  fee: Fee;
+}
+
+export class ValidatorList extends Struct {
+  accountType: AccountType;
+  maxValidators: number;
+  validators: [ValidatorStakeInfo];
+}
+export class ValidatorStakeInfo extends Struct {
+  status: StakeStatus;
+  voteAccountAddress: PublicKey;
+  stakeLamports: BN;
+  lastUpdateEpoch: BN;
+}
+export class StakeStatus extends Enum {}
+
+export class StakeStatusEnum extends Struct {}
+
+export enum StakeStatusKind {
+  Active = 'Active',
+  DeactivatingTransient = 'DeactivatingTransient',
+  ReadyForRemoval = 'ReadyForRemoval',
+}
+
+export class PublicKey extends Struct {
+  value: BN;
+}
+
+export const SCHEMA: Schema = constructStakePoolSchema(SOLANA_SCHEMA);
+
+export function constructStakePoolSchema(SCHEMA: Schema): Schema {
```

Comment:
nit, all caps variable names are typically globals
```suggestion
export function constructStakePoolSchema(schema: Schema): Schema {
```